data/src/main/scala/cats/data/Or.scala

```diff
@@ -1,498 +1,152 @@
 package cats
 package data
 
-import algebra.{Eq, Monoid, Order, Semigroup}
+import Or.{LeftOr, RightOr}
+import algebra.{PartialOrder, Eq, Order, Monoid}
+import cats.functor.Invariant
 
-import scala.util.control.NonFatal
-
-/* Represents a disjunction: a result that is either an `A` or a `B`.
+/** Represents a right-biased disjunction that is either an `A` or a `B`.
  *
- * An instance of `A` [[Or]] B is either a [[LOr]]`[A]` (aka a "left") or a [[ROr]]`[B]` (aka a "right").
+ * An instance of `A` [[Or]] `B` is either a [[LeftOr]]`[A]` or a [[RightOr]]`[B]`.
  *
- * A common use of a disjunction is to explicitly represent the possibility of failure in a result as opposed to
- * throwing an exception. By convention, the left is used for errors and the right is reserved for successes.
+ * A common use of [[Or]] is to explicitly represent the possibility of failure in a result as opposed to
+ * throwing an exception.  By convention, [[LeftOr]] is used for errors and [[RightOr]] is reserved for successes.
  * For example, a function that attempts to parse an integer from a string may have a return type of
  * `NumberFormatException` [[Or]] `Int`. However, since there is no need to actually throw an exception, the type (`A`)
  * chosen for the "left" could be any type representing an error and has no need to actually extend `Exception`.
  *
  * `A` [[Or]] `B` is isomorphic to `scala.Either[A, B]`, but [[Or]] is right-biased, so methods such as `map` and
  * `flatMap` apply only in the context of the "right" case. This right bias makes [[Or]] more convenient to use
- * than `scala.Either` in a monadic context. Methods such as `swap`, `swapped`, and `leftMap` provide functionality
+ * than `scala.Either` in a monadic context. Methods such as `swap`, and `leftMap` provide functionality
  * that `scala.Either` exposes through left projections.
  */
 sealed abstract class Or[+A, +B] extends Product with Serializable {
-  /*
-  final class SwitchingOr[X](r: => X) {
-    def <<?:(left: => X): X =
-      Or.this match {
-        case LOr(_) => left
-        case ROr(_) => r
-      }
+  def fold[C](fa: A => C, fb: B => C): C = this match {
+    case RightOr(b) => fb(b)
```

Comment:
it is really a details but I think it reads slightly better if you pattern match first with `LeftOr` since the first function is on A